### DISCUSSION TITLE
dbt-sqlserver 1.7.4 is a disaster! isn't it?

### DISCUSSION THREAD

**Discussion by MosBabe** [dbt-msft/dbt-sqlserver#503] [Q&A]
We were using dbt-sqlserver 1.4.3 before we update this adapter to 1.7.4 .

When we run an asset in dbt-core, dbt-sqlserver 1.4.3 adapter first create a temp table and try to fill this temp table regarding to our sql statement. After it completes filling the temp table, drop the main table and rename the temp table as main table. 

dbt-sqlserver 1.7.4 adapter first rename the main table like "maintable__dbt_backup" then try to fill main table with my sql statement. While it is filling the main table, main table is inaccessible/unreachable. This is the problem 1. 
If I terminate running the sql statement while it is trying to fill the main table, Then I can not find the main table.  When I try to access the table ("SELECT * FROM main table") , I encounter the INVALID OBJECT problem. This is the problem 2.

We change our system from Microsoft SSIS to dbt and dagster. It means from old(obsolote) data stack to modern data stack. But this dbt-sqlserver 1.7.4 adapter seems like insist on bringing obsolote data stacks back

I wonder If anybody encounter these problems? 
Thank you in advance.

**Comment by joegood** [comment]
I'm in the process of doing that same migration, 1.4 -> 1.7.  There have been so many other issues with moving to the newer version that I haven't even had a chance to see how the SQL actually runs.  If this is true, that is going to be a problem for us.

**Reply by MosBabe** [reply]
After the migration; you can change the files that I wrote below with the same files from 1.4.3 , then try to adapt it to your system or code. It works for us.

*file name : create_table_as.sql      directory: .venv\Lib\site-packages\dbt\include\sqlserver\macros\materializations\models\table
*file name : relation.sql                  directory: .venv\Lib\site-packages\dbt\include\sqlserver\macros\adapters
*file name : table.sql                      directory: .venv\Lib\site-packages\dbt\include\fabric\macros\materializations\models\table

ATTENTION!: We do not use fabric, so you have to check the files that I mentioned.

Please control this commit before you change something: https://github.com/dbt-msft/dbt-sqlserver/commit/e719d7185bb6700355ec5eb14fb4ad9d73b97e3f
This commit destroy the experince on SQL SERVER. They probably focused on fabric!

**Comment by cody-scott** [comment]
Adding some comments here that its been fine for our use cases. Are you still having those issues? I had put the original changes through with the hope that others would pick up the continued support of it.

**Reply by cody-scott** [reply]
One additional comment is the test coverage is poor with this. It was mentioned elsewhere as well as a point of improvement if people are looking for work.

I'm not a maintainer on the project so we're all waiting on the review from others.

**Comment by cody-scott** [comment]
@MosBabe  any chance you could give some feedback if the 1.8.0rc1 helps with the issues you're seeing?

**Reply by cody-scott** [reply]
1.8 is out that should fix most of the issues from the 1.7 version. 

Closing this but feel free to open a new issue or discussion if there is missed regressions.
